.travis.yml

```diff
@@ -18,7 +18,36 @@ jobs:
       services:
         - docker
       env:
-        - CODALAB_USERNAME=codalab CODALAB_PASSWORD=codalab CI=true
+        - CODALAB_USERNAME=codalab CODALAB_PASSWORD=codalab CI=true CODALAB_STATE_BACKEND=json
+      python: 3.6
+      before_install:
+        - sudo rm /usr/local/bin/docker-compose
+        - curl -L https://github.com/docker/compose/releases/download/1.23.2/docker-compose-`uname -s`-`uname -m` > docker-compose
+        - chmod +x docker-compose
+        - sudo mv docker-compose /usr/local/bin
+        - sudo service mysql stop
+      install:
+        - python3 codalab_service.py build services --version ${TRAVIS_BRANCH} --pull $([ -z "${CODALAB_DOCKER_USERNAME}" ] || [ "$TRAVIS_PULL_REQUEST_BRANCH" != "" ] || echo "--push")
+        - python3 codalab_service.py start --services default monitor --version ${TRAVIS_BRANCH}
+        - pip install -r requirements.txt
+      before_script:
+        - ./scripts/test-setup.sh
+        - export PATH=$PATH:/usr/bin/geckodriver
+      script:
+        - python3 test_runner.py --version ${TRAVIS_BRANCH} default
+        # Make sure restarting worker doesn't cause any issues (ie in serialization/deserialization)
+        - docker restart codalab_worker_1 && python3 test_runner.py --version ${TRAVIS_BRANCH} run
+        - docker stop codalab_worker_1
+        - python3 codalab_service.py start --services init worker --version ${TRAVIS_BRANCH} --shared-file-system
+        # Run frontend tests + backend test modules that test shared file systems code paths
+        - python3 test_runner.py --version ${TRAVIS_BRANCH} frontend basic run read write kill resources netcat netcurl workers
```

Comment:
Do we really need to run _all_ these tests again? What do you think a minimal set of tests would be (just to minimize additional CI build time)?

Reply:
yeah, I was being a bit conservative here just to ensure that things worked properly. In practice, when I was developing, I didn't see travis build times actually being that impacted. Everything runs in parallel regardless, and the time for each of the backends is basically identical. I know you've been working on getting things moved to Actions, do you think it would be more of a problem there?

> What do you think a minimal set of tests would be (just to minimize additional CI build time)?

I'm honestly not sure, I'm not too familiar with the test suite—your input would be valuable here. At various times when I was working on this PR, I had tests failing in each stage (frontend tests, shared fs tests, the regular tests), so they're all somewhat useful imo...

Reply:
Is there a way to avoid all the config duplication with the two versions?

Reply:
I tried looking for something, but I didn't see anything that would work without dramatically reworking the way the tests are set up (e.g., making a meta single test script that runs an appropriate check depending on an environment variable).